src/extractor/guidance/mergable_road_detector.cpp

```diff
@@ -0,0 +1,427 @@
+#include "extractor/guidance/mergable_road_detector.hpp"
+#include "extractor/guidance/node_based_graph_walker.hpp"
+
+#include "util/bearing.hpp"
+#include "util/guidance/name_announcements.hpp"
+
+using osrm::util::angularDeviation;
+
+namespace osrm
+{
+namespace extractor
+{
+namespace guidance
+{
+
+namespace
+{
+const auto ASSUMED_LANE_WIDTH = 3.25;
+const auto MERGABLE_ANGLE_DIFFERENCE = 95.0;
+
+// check a connected road for equality of a name
+inline auto makeCheckRoadForName(const NameID name_id,
+                                 const util::NodeBasedDynamicGraph &node_based_graph)
+{
+    return [name_id, &node_based_graph](const MergableRoadDetector::MergableRoadData &road) {
+        // since we filter here, we don't want any other name than the one we are looking for
+        return name_id != node_based_graph.GetEdgeData(road.eid).name_id;
```

Comment:
Hm so this is doing the name_id check and not the name change heuristic check - why?